# Programming Exercises

1.(Convert Celsius to Fahrenheit) Write a program that reads a Celsius degree from
the console and converts it to Fahrenheit and displays the result. The formula for
the conversion is as follows:

    fahrenheit = (9 / 5) * celsius + 32

In [1]:
# Takes in the celsius value from user
celsius = input("Enter Degree in Celsius:")

# Computes the value of fahrenheit
fahrenheit = (9 / 5) * int(celsius) + 32

#Displays the result
print("{} Celsius is {} Fahrenheit".format(celsius, fahrenheit))

Enter Degree in Celsius:12
12 Celsius is 53.6 Fahrenheit


2.(Compute the volume of a cylinder) Write a program that reads in the radius and
length of a cylinder and computes the area and volume using the following formulas:

    area = radius x radius x π

    volume = area * length

In [8]:
# Takes in the Radius and Length
radius, lenght = eval(input("Enter Radius and Lenght of the Cylinder:"))

# Declaring a constant Pi value
PI = 3.14159

# Computes The Area and Volume
area = radius * radius * PI
volume = area * lenght

# Displays The Result
print("The Area Is", area)
print("The Volume Is",volume)

Enter Radius and Lenght of the Cylinder:5.5, 12
The Area Is 95.0330975
The Volume Is 1140.39717


3. (Convert feet into meters) Write a program that reads a number in feet, converts it
to meters, and displays the result. One foot is 0.305 meters.

In [2]:
# Getting feet value from the user
feet = eval(input("Enter a value for feet:"))

# Converting feet to meters
meters = feet * 0.305

# Displaying Result
print("{} Feet is {} Meters".format(feet, meters))

Enter a value for feet:89
89 Feet is 27.145 Meters


4. (Convert pounds into kilograms) Write a program that converts pounds into
kilograms. The program prompts the user to enter a value in pounds, converts it to
kilograms, and displays the result. One pound is 0.454 kilograms.

In [3]:
# Prompting the user to enter value in pounds
pounds = eval(input("Enter a value in pounds:"))

# Converting pounds to kilograms
kilograms = pounds * 0.454

#Displays the result
print("{} Pounds is {} Kilograms".format(pounds, kilograms))

Enter a value in pounds:55.5
55.5 Pounds is 25.197 Kilograms


*5.(Financial application: calculate tips) Write a program that reads the subtotal and
the gratuity rate and computes the gratuity and total. For example, if the user
enters 10 for the subtotal and 15% for the gratuity rate, the program displays 1.5
as the gratuity and 11.5 as the total.

In [4]:
# Getting Subtotal and gratuity rate from the user
subtotal, gratuity_rate = eval(input("Enter The Subtotal and a Gratuity rate:"))

# Computing Gratuity and total amount
gratuity = (gratuity_rate / 100) * subtotal
total = subtotal + gratuity

# Displaying the results
print("The Gratuity is {} and the total is {}".format(gratuity,total))

Enter The Subtotal and a Gratuity rate:15.69, 15
The Gratuity is 2.3535 and the total is18.043499999999998


**6. (Sum the digits in an integer) Write a program that reads an integer between 0 and
1000 and adds all the digits in the integer. For example, if an integer is 932, the
sum of all its digits is 14. (Hint: Use the % operator to extract digits, and use the //
operator to remove the extracted digit. For instance, 932 % 10 = 2 and 932 //
10 = 93.)

In [5]:
# Getting a digit between 0 and 1000 from the user
digits = eval(input("Enter a digit between 0 and 1000:"))

# Getting digits one by one
first_number = digits // 100
second_number = (digits // 10) % 10
third_number = digits % 10

# Summing up the digits
sum_of_digits = first_number + second_number + third_number

# Displaying the result
print("The Digits are {} {} {}".format(first_number,second_number,third_number))
print("Their Sum Is: {}".format(sum_of_digits))

Enter a digit between 0 and 1000:932
The Digits are 9 3 2
Their Sum Is: 14


**7. (Find the number of years and days) Write a program that prompts the user to
enter the minutes (e.g., 1 billion), and displays the number of years and days for
the minutes. For simplicity, assume a year has 365 days.

In [1]:
#Taking values from user
minutes = eval(input("Enter the Minutes:"))

# Computing number of Years and Days
total_minutes_per_day = 60 * 24
minutes_per_year = total_minutes_per_day * 365
number_of_years = minutes // minutes_per_year
days = minutes % minutes_per_year // 1440

#Displaying Results
print("{} Minutes is Approximately {} Years and {} days".format(minutes, number_of_years,days))

Enter the Minutes:1000000000
1000000000 Minutes is Approximately 1902 Years and 214 days


8.(Science: calculate energy) Write a program that calculates the energy needed to
heat water from an initial temperature to a final temperature. Your program should 
prompt the user to enter the amount of water in kilograms and the initial and final
temperatures of the water. The formula to compute the energy is

    Q = M x (finalTemperature – initialTemperature) x 4184

where M is the weight of water in kilograms, temperatures are in degrees Celsius,
and energy Q is measured in joules.

In [1]:
# Taking Values from the user
amount_of_water = eval(input("Enter the amount of water in kilograms:"))
initial_temperature = eval(input("Enter The Initial Temperature:"))
final_temperature = eval(input("Enter the Final Temperature:"))

# Computing the energy required to heat the water
energy = amount_of_water * (final_temperature - initial_temperature) * 4184

# Displaying Results
print("The Energy Needed Is:{}".format(energy))

Enter the amount of water in kilograms:55.5
Enter The Initial Temperature:3.5
Enter the Final Temperature:10.5
The Energy Needed Is:1625484.0


*9.(Science: wind-chill temperature) How cold is it outside? The temperature alone is
not enough to provide the answer. Other factors including wind speed, relative
humidity, and sunshine play important roles in determining coldness outside. In
2001, the National Weather Service (NWS) implemented the new wind-chill temperature
to measure the coldness using temperature and wind speed. The formula
is given as follows:

    twc = 35.74 + 0.6215ta - 35.75v^0.16 + 0.4275ta v^0.16
    
where ta is the outside temperature measured in degrees Fahrenheit and v is the
speed measured in miles per hour.twc is the wind-chill temperature. The formula
cannot be used for wind speeds below 2 mph or for temperatures below -58°F or
above 41°F.
Write a program that prompts the user to enter a temperature between -58°F and
41°F and a wind speed greater than or equal to 2 and displays the wind-chill temperature.

In [2]:
# Taking Values From The User
temperature = eval(input("Enter the temperature in fahrenheit between -58 and 41:"))
wind_speed = eval(input("Enter the wind speed(above 2) in miles per hour:"))

# Computing the Wind-chill Temperature
wind_chill_temperature = 35.74 + 0.6215 * temperature - 35.75 * wind_speed ** 0.16 + 0.4275 * temperature * wind_speed ** 0.16

# Displaying Results
print("The Wind Chill index is:{}".format(wind_chill_temperature))

Enter the temperature in fahrenheit between -58 and 41:5.3
Enter the wind speed(above 2) in miles per hour:6
The Wind Chill index is:-5.567068455881625


*10.(Physics: find runway length) Given an airplane’s acceleration a and take-off
speed v, you can compute the minimum runway length needed for an airplane to
take off using the following formula: 

    Lenght = (v x v) / 2 x a.

Write a program that prompts the user to enter v in meters/second (m/s) and the
acceleration a in meters/second squared and displays the minimum runway
length.

In [3]:
# Prompting User for inputs
speed, acceleration = eval(input("Enter the Speed and acceleration:"))

# Computing the runway length
lenght = (speed ** 2) / (2 * acceleration)

# Display Results
print("The minimum runway length for this airplane is {}".format(lenght))

Enter the Speed and acceleration:60, 3.5
The minimum runway length for this airplane is 514.2857142857143


*11.(Financial application: investment amount) Suppose you want to deposit a
certain amount of money into a savings account with a fixed annual interest rate.
What amount do you need to deposit in order to have $5,000 in the account after
three years? The initial deposit amount can be obtained using the following
formula:

    initialDepositAmount = finalAccountValue / (1 + monthlyInterestRate)^numberOfMonths
    
Write a program that prompts the user to enter final account value, annual interest
rate in percent, and the number of years, and displays the initial deposit amount.

In [4]:
# Prompting User of inputs
final_account_value = eval(input("Enter final account value:"))
annual_interest_rate = eval(input("Enter annual interest rate in percent:"))
number_of_years = eval(input("Enter Number Of Years:"))

# Computing the Investment Amount
monthly_interest_rate = annual_interest_rate / 1200
number_of_months = number_of_years * 12
initial_deposit_amount = final_account_value / (1 + monthly_interest_rate) ** number_of_months

# Displaying results
print("Initial deposit value is {}".format(initial_deposit_amount))


Enter final account value:1000
Enter annual interest rate in percent:4.25
Enter Number Of Years:5
Initial deposit value is 808.8639197424636


12.(Print a table) Write a program that displays the following table:

    a    b    a ** b
    1    2     1
    2    3     8
    3    4     81
    4    5     1024
    5    6     15625

In [32]:
# Displaying the result 
print("a \t b \t a**b")
print("1 \t 2 \t 1")
print("2 \t 3 \t 8")
print("3 \t 4 \t 81")
print("4 \t 5 \t 1024")
print("5 \t 6 \t 15625")

a 	 b 	 a**b
1 	 2 	 1
2 	 3 	 8
3 	 4 	 81
4 	 5 	 1024
5 	 6 	 15625


*13.(Split digits) Write a program that prompts the user to enter a four-digit integer
and displays the number in reverse order.

In [5]:
# Prompting User for input
four_digit_number = eval(input("Enter a four digit number:"))

# Splitting digits
first_digit = four_digit_number // 1000
second_digit = (four_digit_number // 100) % 10
third_digit = (four_digit_number % 100) // 10
fourth_digit = (four_digit_number % 100) % 10

# Displaying results
print("The Numbers Are: {} {} {} {}".format(first_digit,second_digit,third_digit,fourth_digit))

Enter a four digit number:3125
The Numbers Are: 3 1 2 5


*14.(Geometry: area of a triangle) Write a program that prompts the user to enter the
three points (x1, y1), (x2, y2), and (x3, y3) of a triangle and displays its area.
The formula for computing the area of a triangle is

    s = (side1 + side2 + side3) / 2

    area = (s(s - side1)(s - side2)(s - side3)) ** 0.5

In [5]:
# Get user input
x1, y1, x2, y2, x3, y3 = eval(input("Enter three points for a triangle:"))

# Calculate the sides
side1 = ((x1 - x2) ** 2 + (y1 - y2) ** 2) ** 0.5
side2 = ((x1 - x3) ** 2 + (y1 - y3) ** 2) ** 0.5
side3 = ((x2 - x3) ** 2 + (y2 - y3) ** 2) ** 0.5

# compute s from sides and calculate area
s = (side1 + side2 + side3) / 2
area = (s * (s - side1) * (s - side2) * (s - side3)) ** 0.5

# Displaying the results
print(f"The Area of the Triangle is {area}")


Enter three points for a triangle:1.5, -3.4, 4.6, 5, 9.5, -3.4
The Area of the Triangle is 33.600000000000016


15.(Geometry: area of a hexagon) Write a program that prompts the user to enter the
side of a hexagon and displays its area. The formula for computing the area of a
hexagon is

    Area = (3 * (3)**0.5 / 2) * s ** 2, 
 where s is the length of a side.

In [6]:
# Prompting User for input values
sides = eval(input("Enter the side:"))

# Calculating the area
area = (3 * (3) ** 0.5 / 2) * sides ** 2

# Displaying the results 
print("The area of the hexagon is {}".format(area))

Enter the side:5.5
The area of the hexagon is 78.59180539343781


16.(Physics: acceleration) Average acceleration is defined as the change of velocity
divided by the time taken to make the change, as shown in the following formula:

    a = (vf - vi) / t
    
Write a program that prompts the user to enter the starting velocity vi in
meters/second, the ending velocity vf in meters/second, and the time span t in
seconds, and displays the average acceleration.

In [8]:
# Prompting user for input values
initial_velocity, final_velocity = eval(input("Enter the initial and final velocity in meters:"))
time_taken = eval(input("Enter the time taken in seconds:"))

# Calculating acceleration
acceleration = (final_velocity - initial_velocity) / time_taken

# Displaying the average acceleration
print("The Average Acceleration is {}".format(acceleration))

Enter the initial and final velocity in meters:5.5, 50.9
Enter the time taken in seconds:4.5
The Average Acceleration is 10.088888888888889


*17.(Health application: compute BMI) Body mass index (BMI) is a measure of health
based on weight. It can be calculated by taking your weight in kilograms and
dividing it by the square of your height in meters. Write a program that prompts
the user to enter a weight in pounds and height in inches and displays the BMI.
Note that one pound is 0.45359237 kilograms and one inch is 0.0254 meters.

In [9]:
# Prompting User for input values
weight_in_pounds = eval(input("Enter Weight in pounds:"))
height_in_inches = eval(input("Enter Height in inches:"))

# Computing BMI
weight_in_kilogram = weight_in_pounds * 0.45359237
height_in_meters = height_in_inches * 0.0254

BMI = weight_in_kilogram / height_in_meters ** 2

# Displaying the BMI Result
print("BMI is:{}".format(BMI))

Enter Weight in pounds:95.5
Enter Height in inches:50
BMI is:26.857257942215885


*18.(Current time) Create a program that displays the
current time in GMT. Revise the program so that it prompts the user to enter the
time zone in hours away from (offset to) GMT and displays the time in the specified
time zone.

In [12]:
import time
# Prompting user for Time zone Input value
timeZone = eval(input("Enter the time zone Offset to GMT:"))

# Getting Current time
currentTime = time.time()

# Getting Total Seconds
totalSeconds = int(currentTime)

# get current seconds
currentSecond = totalSeconds % 60

# obtaining total minutes
totalMinutes = totalSeconds // 60

# get current minutes
currentMinute = totalMinutes % 60

# obtaining total hours
totalHours = totalMinutes // 60

# get current hour
currentHour = totalHours % 24

# Obtaining GMT Offset time zone
currentHour = currentHour + timeZone

# Displaying Current time
print("The Current Time is -:- {} : {} : {} ".format(currentHour,currentMinute,currentSecond))

Enter the time zone Offset to GMT:1
The Current Time is -:- 14 : 6 : 49 


*19.(Financial application: calculate future investment value) Write a program that
reads in an investment amount, the annual interest rate, and the number of years,
and displays the future investment value using the following formula:

    futureInvestmentValue = investmentAmount * (1 + monthlyInterestRate)^ numberOfMonths
    
For example, if you enter the amount 1000, an annual interest rate of 4.25%,
and the number of years as 1, the future investment value is 1043.33

In [13]:
# Reading Invesment amount from the user
investment_amount = eval(input("Enter Investment Amount:"))
annual_interest_rate = eval(input("Enter Annual Interest rate:"))
number_of_years = eval(input("Enter Number of years:"))

#Calculating the future investment value
monthly_interest_rate = annual_interest_rate / 1200
number_of_months = number_of_years * 12
future_investment_value = investment_amount * (1 + monthly_interest_rate) ** number_of_months

# Displaying The future investment 
print("Accumulated value is: {}".format(future_investment_value))

Enter Investment Amount:1000
Enter Annual Interest rate:4.25
Enter Number of years:1
Accumulated value is: 1043.337716309617


**20.(Financial application: calculate interest) If you know the balance and the annual
percentage interest rate, you can compute the interest on the next monthly payment
using the following formula:

    interest = balance * (annualInterestRate / 1200)

Write a program that reads the balance and the annual percentage interest rate
and displays the interest for the next month.

In [14]:
# Prompting user for input values
balance,interest_rate = eval(input("Enter The balance and interest rate(e.g 3 for 3%):"))

# calculating interest
interest = balance * (interest_rate / 1200)

# Displaying the interest as the value
print("The Interest is: {}".format(interest))

Enter The balance and interest rate(e.g 3 for 3%):1000,3.5
The Interest is: 2.916666666666667


**21.(Financial application: compound value) Suppose you save $100 each month into
a savings account with an annual interest rate of 5%.Therefore, the monthly interest
rate is 0.05 / 12 = 0.00417. After the first month, the value in the account
becomes

    100 * (1 + 0.00417) = 100.417
    
After the second month, the value in the account becomes

    (100 + 100.417) * (1 + 0.00417) = 201.252
    
After the third month, the value in the account becomes

    (100 + 201.252) * (1 + 0.00417) = 302.507

and so on.

Write a program that prompts the user to enter a monthly saving amount and
displays the account value after the sixth month. Here is a sample run of the
program:


In [15]:
# Prompting user for monthly saving amount
monthly_saving_amount = eval(input("Enter monthy saving amount:"))

# Computing the compound value
first_month = 100 * (1 + 0.00417)
second_month = (100 + first_month) * (1 + 0.00417)
third_month = (100 + second_month) * (1 + 0.00417)
fourth_month = (100 + third_month) * (1 + 0.00417)
fifth_month = (100 + fourth_month) * (1 + 0.00417)
sixth_month = (100 + fifth_month) * (1 + 0.00417)

# Displaying results
print("After the sixth month, the account value is {}".format(sixth_month))

Enter monthy saving amount:100
After the sixth month, the account value is 608.8181155768638


22.(Population projection) The US Census Bureau projects population based on the
following assumptions:

    One birth every 7 seconds
    One death every 13 seconds
    One new immigrant every 45 seconds


Write a program to display the population for each of the next five years. Assume the
current population is 312032486 and one year has 365 days. Hint: in Python, you
can use integer division operator // to perform division. The result is an integer. For
example, 5 // 4 is 1 (not 1.25) and 10 // 4 is 2 (not 2.5). Make the program to prompt the user to enter the number of years and displays the population after that many years.

In [16]:
# Prompting user for input value
number_of_years = eval(input("Enter the Number of years:"))

#Calculating the population 
total_seconds_per_year = 60 * 60 * 24 * 365
total_births_per_year = total_seconds_per_year // 7
total_deaths_per_year = total_seconds_per_year // 13
total_immigrants_per_year = total_seconds_per_year // 45
current_population = 312032486
new_population = current_population + ((total_births_per_year + total_immigrants_per_year - total_deaths_per_year) * number_of_years)

# Displaying results
print("The Population in 5 years is {}".format(new_population))

Enter the Number of years:5
The Population in 5 years is 325932966


# End Of Exercises